src/auth/oauth2/core/qgsauthoauth2config.cpp

```diff
@@ -230,6 +231,14 @@ void QgsAuthOAuth2Config::setCustomHeader( const QString &header )
     emit customHeaderChanged( mCustomHeader );
 }
 
+void QgsAuthOAuth2Config::setExtraTokens( const QVariantMap &tokens )
+{
+  const QVariantMap preval( mExtraTokens );
+  mExtraTokens = tokens;
+  if ( preval != tokens )
+    emit extraTokensChanged( mExtraTokens );
```

Comment:
A bit cleaner:

```suggestion
  if ( tokens == mExtraTokens )
    return;
    
  mExtraTokens = tokens;
  emit extraTokensChanged( mExtraTokens );
```

Reply:
@nyalldawson , yes, not a bit cleaner, _a ton more cleaner_. The only reason I didn't do that was to match the style with every other function. If you don't mind the difference, then sure :)